### PR TITLE
Format inner attributes

### LINKED ISSUE #4206: rustfmt ignores #![rustfmt::skip] when run with -r
Given the following project structure,

```
.
├── Cargo.lock
├── Cargo.toml
└── src
    ├── foo.rs
    └── main.rs
```

```rust
// main.rs
mod foo;

fn main() {
    println!("Hello, world!");
}
```
```rust
// foo.rs
#![rustfmt::skip]

fn
foo()
{}
```

`rustfmt src/main.rs -c` and `rustfmt src/foo.rs -c` succeeds but `rustfmt -cr src/main.rs` will fail with the following output:

```diff
Diff in /tmp/skip/src/foo.rs:1:
 #![rustfmt::skip]
 
-fn
-foo()
-{}
+fn foo() {}
 
```

### PR DESCRIPTION
**Pull request by topecongiro**
Close #4206.

### CODE REVIEW

File: rustfmt-core/rustfmt-lib/src/formatting/modules.rs [hunk 1]
```diff
@@ -178,7 +220,6 @@ impl<'ast, 'sess, 'c> ModResolver<'ast, 'sess> {
     fn insert_sub_mod(
         &mut self,
         sub_mod_kind: SubModKind<'c, 'ast>,
-        _sub_mod: Cow<'ast, ast::Mod>,
```
Comment (calebcartwright) [inline rustfmt-core/rustfmt-lib/src/formatting/modules.rs:181]: Oops! I made this change during a previous rustc-ap-* round to get the code compiling again and forgot to remove this after :man_facepalming:

File: rustfmt-core/rustfmt-lib/src/formatting/syntux/parser.rs [hunk 2]
```diff
@@ -102,79 +102,15 @@ impl<'a> Parser<'a> {
...
-    ) -> Result<ast::Mod, ParserError> {
+    ) -> Result<(ast::Mod, Vec<ast::Attribute>), ParserError> {
         let result = catch_unwind(AssertUnwindSafe(|| {
             let mut parser = new_parser_from_file(sess.inner(), &path, Some(span));
-
-            let lo = parser.token.span;
-            // FIXME(topecongiro) Format inner attributes (#3606).
-            match Parser::parse_inner_attrs(&mut parser) {
-                Ok(_attrs) => (),
+            match parser.parse_mod(&TokenKind::Eof) {
```
Comment (calebcartwright) [inline rustfmt-core/rustfmt-lib/src/formatting/syntux/parser.rs:112]: much simpler! :heart:

File: rustfmt-core/rustfmt-lib/src/formatting/syntux/session/ignore_path.rs [hunk 3]
```diff
@@ -8,7 +10,11 @@ pub(crate) struct IgnorePathSet {
 
 impl IgnorePathSet {
     pub(crate) fn from_ignore_list(ignore_list: &IgnoreList) -> Result<Self, ignore::Error> {
-        let mut ignore_builder = GitignoreBuilder::new(ignore_list.rustfmt_toml_path());
+        let root = ignore_list
+            .rustfmt_toml_path()
+            .parent()
```
Comment (calebcartwright) [inline rustfmt-core/rustfmt-lib/src/formatting/syntux/session/ignore_path.rs:15]: why the parent here?

Comment (topecongiro) [reply rustfmt-core/rustfmt-lib/src/formatting/syntux/session/ignore_path.rs:15]: Looking at [this CI failure](https://travis-ci.com/github/rust-lang/rustfmt/jobs/339857177), I noticed that we are ought to pass a directory to [`GitignoreBuilder::new`](https://docs.rs/ignore/0.4.15/ignore/gitignore/struct.GitignoreBuilder.html#method.new), hence this change.

**Pull request merged**